### PR TITLE
[MIG][9.0] medical_lab: Migrate oemedical_lab

### PR DESCRIPTION
**Pull request by lasley**
I primarily did this upgrade due to the licensing dispute in #133 & the fact that people were paying 500 EU to use a migrated version of this code. I figured might as well fill the product gap that was opened with the removal of the infringing app from the store.

* Upgrade oemedical_lab to v9

Depends:
- [x] #138

### CODE REVIEW

**Review by susport — COMMENTED** [review]
Why in runbot only see the menu Medical - Patient?

**Comment by lasley** [comment]
Good catch @susport - that's an issue with the Travis file (#138). I installed the module manually and it seems to work fine.

**Comment by lasley** [comment]
I rebased this branch onto the Travis fix in #138 so we can have accurate test results. Will rebase again once its merged.

**Review by susport — COMMENTED** [review]
You have a problem with the test according in travis error

**Comment by lasley** [comment]
Thanks @susport - Testing issue should be fixed up once #142 is merged & I rebase.

**Comment by lasley** [comment]
All Labs=> Create
```
  File "/root/build/OCA/vertical-medical/medical_lab/models/medical_lab.py", line 30, in <lambda>
    default=lambda s: s._default_physician_id(),
AttributeError: 'medical.lab' object has no attribute '_default_physician_id'
```

File: medical_lab/models/medical_lab_test_type.py
Comment (susport) [inline medical_lab/models/medical_lab_test_type.py:23]: Here just list all the products, for this case can you filter only the types of products that are services or another that is defined?

**Comment by lasley** [comment]
@t3ddftw please review

File: medical_lab/__openerp__.py
Comment (tedsalmon) [inline medical_lab/__openerp__.py]: `s/coding=utf-8/# -*- coding: utf-8 -*-`

Lint is complaining about this too.

File: medical_lab/models/medical_lab.py
Comment (tedsalmon) [inline medical_lab/models/medical_lab.py]: `s/coding=utf-8/# -*- coding: utf-8 -*-`

Lint is complaining about this too.

File: medical_lab/models/medical_lab_patient.py
Comment (tedsalmon) [inline medical_lab/models/medical_lab_patient.py]: `s/coding=utf-8/# -*- coding: utf-8 -*-`

Lint is complaining about this too.

File: medical_lab/models/medical_lab_test_criterion.py
Comment (tedsalmon) [inline medical_lab/models/medical_lab_test_criterion.py]: `s/coding=utf-8/# -*- coding: utf-8 -*-`

Lint is complaining about this too.

File: medical_lab/models/medical_lab_test_type.py
Comment (tedsalmon) [inline medical_lab/models/medical_lab_test_type.py]: `s/coding=utf-8/# -*- coding: utf-8 -*-`

Lint is complaining about this too.

**Review by tedsalmon — APPROVED** [review]
Minor changes requested. Otherwise, LGTM! 👍

**Comment by lasley** [comment]
@susport @t3ddftw - thanks for the reviews; all points fixed

**Comment by lasley** [comment]
@bmw95 please review

File: medical_lab/README.rst
Comment (woodbrettm) [inline medical_lab/README.rst]: Remove line 36-39

File: medical_lab/README.rst
Comment (woodbrettm) [inline medical_lab/README.rst]: Remove line 47

File: medical_lab/README.rst
Comment (woodbrettm) [inline medical_lab/README.rst:50]: Remove line 55

File: medical_lab/README.rst
Comment (woodbrettm) [inline medical_lab/README.rst:53]: Missing **Images** section here

File: medical_lab/README.rst
Comment (woodbrettm) [inline medical_lab/README.rst]: smashing => smash

File: medical_lab/__init__.py
Comment (woodbrettm) [inline medical_lab/__init__.py:1]: Missing Copyrights + license

File: medical_lab/__openerp__.py
Comment (woodbrettm) [inline medical_lab/__openerp__.py:6]: https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py

Ordered improperly + mix of single and double quotes. Should all be double.

File: medical_lab/models/__init__.py
Comment (woodbrettm) [inline medical_lab/models/__init__.py:1]: Missing copyrights + license

File: medical_lab/models/medical_lab.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab.py:28]: Either change to physician, or change **request_physician_id** string attribute to 'requesting pathologist' instead of requesting physician.

File: medical_lab/models/medical_lab.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab.py:30]: Same as previous comment

File: medical_lab/models/medical_lab.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab.py]: missing comma at end of attribute

File: medical_lab/models/medical_lab_patient.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab_patient.py]: I've been doing the previous migrations with selection fields done as follows to keep in line with other attributes better. Up to you of course, but just pointing out in case of wanting to keep consistency :)
````python
state = fields.Selection(
    selection=[
        ...
    ],
)
````

File: medical_lab/models/medical_lab_test_criterion.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab_test_criterion.py:10]: Remove line 10

File: medical_lab/models/medical_lab_test_criterion.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab_test_criterion.py]: Test Types

File: medical_lab/models/medical_lab_test_criterion.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab_test_criterion.py:16]: Do we need to explicitly define **Test Name** instead of just letting Odoo use **Name**? Not sure. I guess since other fields have 'Test' in their string attributes having Test Name makes sense.

File: medical_lab/models/medical_lab_test_result.py
Comment (woodbrettm) [inline medical_lab/models/medical_lab_test_result.py:10]: Remove line 10

File: medical_lab/security/medical_security.xml
Comment (woodbrettm) [inline medical_lab/security/medical_security.xml:1]: Missing copyrights + license

File: medical_lab/views/medical_lab.xml
Comment (woodbrettm) [inline medical_lab/views/medical_lab.xml]: Missing copyrights + utf encoding + license

File: medical_lab/views/medical_lab_patient.xml
Comment (woodbrettm) [inline medical_lab/views/medical_lab_patient.xml]: Copyrights + utf + license

File: medical_lab/views/medical_lab_test_type.xml
Comment (woodbrettm) [inline medical_lab/views/medical_lab_test_type.xml]: Copyrights + utf + license

File: medical_lab/views/medical_menu.xml
Comment (woodbrettm) [inline medical_lab/views/medical_menu.xml]: Copyrights + license + lower-case utf

**Review by woodbrettm — COMMENTED** [review]
Mostly just nit picky stuff. Other than that, LGTM!

File: medical_lab/models/medical_lab.py
Comment (lasley) [reply medical_lab/models/medical_lab.py:28]: A Pathologist is different from a Physician. This is a medical nuance that was intentional.

Clinical Pathologists typically perform the role of a consultant for Patient Physicians.

In this specific instance, it is the MD performing the lab exam. This is typically different from your standard physician.

File: medical_lab/models/medical_lab_test_criterion.py
Comment (lasley) [reply medical_lab/models/medical_lab_test_criterion.py:10]: PEP 8 actually says to put a line here. As Odoo devs, we disobey this rule a lot. My line in the sand is when it is more just a name, description, or inherit.

File: medical_lab/models/medical_lab_test_criterion.py
Comment (lasley) [reply medical_lab/models/medical_lab_test_criterion.py:16]: It's necessary to provide context. Otherwise the inferred context is "Criterion Name", which doesn't fit as well.

**Comment by lasley** [comment]
Thanks for the review @bmw95 - all points fixed or responded to

File: medical_lab/models/medical_lab_test_criterion.py
Comment (woodbrettm) [reply medical_lab/models/medical_lab_test_criterion.py:16]: I agree yeah. Was brainstorming thoughts but quickly realized what you stated.

File: medical_lab/models/medical_lab_test_criterion.py
Comment (woodbrettm) [reply medical_lab/models/medical_lab_test_criterion.py:10]: Interesting. I always wondered that, as PEP8 like you said states to leave a blank line, however I've seen all (most) underscore prefixed attrs grouped with no space between them and the class declaration in Odoo. So if there's just 1 attr, leave a blank line, and if there's a group of underscored attrs, then no blank line? Or should I just always put a blank line in between?

File: medical_lab/models/medical_lab.py
Comment (woodbrettm) [reply medical_lab/models/medical_lab.py:28]: Oh I see. Apologies for my lack of understanding, that makes more sense now.

File: medical_lab/models/medical_lab_test_criterion.py
Comment (lasley) [reply medical_lab/models/medical_lab_test_criterion.py:10]: You should just always put a blank line really. PEP-008 is king; I'm not sure why I make exceptions sometimes. Probably because nobody calls me out on it /shrug

**Pull request merged**
